src/www/views/servers_view/server_connection_indicator/index.ts

```diff
@@ -11,284 +11,191 @@
   limitations under the License.
 */
 
-import {computed, customElement, property} from "@polymer/decorators";
-import {html, PolymerElement} from "@polymer/polymer";
-import {LegacyElementMixin} from "@polymer/polymer/lib/legacy/legacy-element-mixin";
+import {html, css, LitElement} from "lit";
+import {customElement, property, state} from "lit/decorators.js";
 
 export enum ServerConnectionState {
-  INITIAL = "INITIAL",
-  CONNECTING = "CONNECTING",
-  CONNECTED = "CONNECTED",
-  RECONNECTING = "RECONNECTING",
-  DISCONNECTING = "DISCONNECTING",
-  DISCONNECTED = "DISCONNECTED",
+  INITIAL = "initial",
+  CONNECTING = "connecting",
+  CONNECTED = "connected",
+  RECONNECTING = "reconnecting",
+  DISCONNECTING = "disconnecting",
+  DISCONNECTED = "disconnected",
 }
 
+const ANIMATION_DURATION_MS = 1750;
+const ANIMATION_DELAY_MS = 500;
+const CIRCLE_SIZES = [css`large`, css`medium`, css`small`];
+
 @customElement("server-connection-indicator")
-export class ServerConnectionIndicator extends LegacyElementMixin(PolymerElement) {
-  static ANIMATION_DURATION_MS = 1750;
-
-  static template = html`
-    <style>
-      /* Do not mirror animation for RTL languages */
-      /* rtl:begin:ignore */
-      @keyframes rotate-with-pause {
-        0% {
-          transform: rotate(0deg);
-        }
-        60%,
-        100% {
-          transform: rotate(360deg);
-        }
-      }
-      @keyframes rotate-backward-with-pause {
-        0% {
-          transform: rotate(360deg);
-        }
-        60%,
-        100% {
-          transform: rotate(0deg);
-        }
-      }
+export class ServerConnectionIndicator extends LitElement {
+  @property({attribute: "connection-state"}) connectionState: ServerConnectionState;
```

Comment:
I renamed this to `connection-state` since `state` implies a general application or component-specific state